src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -4160,6 +4160,14 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
                 cndSelNode->Op(2) = nestedCndSel->Op(2);
                 if (nestedOp3->IsMaskZero())
                 {
+                    if (nestedOp3->OperIsConvertMaskToVector())
+                    {
+                        GenTree*       zeroMask   = nestedOp3->AsHWIntrinsic()->Op(1);
+                        NamedIntrinsic zeroMaskId = zeroMask->AsHWIntrinsic()->GetHWIntrinsicId();
+                        assert((zeroMaskId >= NI_Sve_CreateFalseMaskByte) &&
+                               (zeroMaskId <= NI_Sve_CreateFalseMaskUInt64));
+                        BlockRange().Remove(zeroMask);
+                    }
                     BlockRange().Remove(nestedOp3);
```

Comment:
Given that we already have the `nestedOp3->SetUnusedValue()` fallback, can we just remove this code entirely and just unconditionally mark it as unused?

Reply:
Probably. I am not sure if `CheckLIR` will complain about anything. Let me try it.